src/dimension.c

```diff
@@ -1067,14 +1069,67 @@ ts_dimension_info_create_closed(Oid table_relid, Name column_name, int32 num_sli
 	return info;
 }
 
+/* Validate the configuration of an open ("time") dimension */
+static void
+dimension_info_validate_open(DimensionInfo *info)
+{
+	Oid dimtype = info->coltype;
+
+	Assert(info->type == DIMENSION_TYPE_OPEN);
+
+	if (OidIsValid(info->partitioning_func))
+	{
+		if (!ts_partitioning_func_is_valid(info->partitioning_func, info->type, info->coltype))
+			ereport(ERROR,
+					(errcode(ERRCODE_INVALID_FUNCTION_DEFINITION),
+					 errmsg("invalid partitioning function"),
+					 errhint("A valid partitioning function for open (time) dimensions must be "
+							 "IMMUTABLE, "
+							 "take the column type as input, and return an integer or "
+							 "timestamp type.")));
+
+		dimtype = get_func_rettype(info->partitioning_func);
+	}
+
+	info->interval = dimension_interval_to_internal(NameStr(*info->colname),
```

Comment:
nit: don't love the validate function having the side-effect of setting a member. But no big deal.

Reply:
FWIW, this is old code I moved around and is actually prevalent in the validation functions. I think the thing that makes this OK is that this is secondary information that is set and can be considered the "result" of the validation. In other words, the information set is not critical for the validation. 

I guess one could break this up in separate validate and "transform"/process functions, but seems out-of-scope of this PR.